uxarray/core/zonal.py

```diff
@@ -0,0 +1,213 @@
+    face_edges_cart: np.ndarray,
+    face_bounds: np.ndarray,
+    face_data: np.ndarray,
+    constLat_deg: float,
+    is_latlonface=False,
+) -> float:
+    """Helper function for _non_conservative_zonal_mean_constant_latitudes.
+    Calculate the zonal mean of the data at a constant latitude. And if only
+    one face is found, return the data of that face.
+
+    Parameters
+    ----------
+    face_edges_cart : np.ndarray, shape (n_face, n_edge, 2, 3)
+        The Cartesian coordinates of the face edges.
+    bounds : np.ndarray, shape (n_face, 2, 2)
+        The latitude and longitude bounds of the faces in radians.
+    face_data : np.ndarray, shape (..., n_face)
+        The data on the faces.
+    constLat_deg : float
+        The constant latitude in degrees. Expected range is [-90, 90].
+    is_latlonface : bool, optional
+        A flag indicating if the current face is a latitudinal/longitudinal (latlon) face,
+        meaning its edges align with lines of constant latitude or longitude. If `True`,
+        edges are treated as following constant latitudinal or longitudinal lines. If `False`,
+        edges are considered as great circle arcs (GCA). Default is `False`.
+
+    Returns
+    -------
+    float
+        The zonal mean of the data at the constant latitude. If there are no faces whose latitude bounds contain the constant latitude, the function will return `np.nan`.
+    """
+
+    # Get the indices of the faces whose latitude bounds contain the constant latitude
+    constLat_rad = np.deg2rad(constLat_deg)
+    candidate_faces_indices = _get_candidate_faces_at_constant_latitude(
+        face_bounds, constLat_rad
+    )
+
+    # Check if there are no candidate faces,
+    if len(candidate_faces_indices) == 0:
+        # Return NaN if there are no candidate faces and raise a warning saying no candidate faces found at this latitude
+        warnings.warn(
+            f"No candidate faces found at the constant latitude {constLat_deg} degrees."
+        )
+        return np.nan
+
+    # Get the face data of the candidate faces
+    candidate_face_data = face_data[..., candidate_faces_indices]
+
+    # Get the list of face polygon represented by edges in Cartesian coordinates
+    candidate_face_edges_cart = face_edges_cart[candidate_faces_indices]
+    candidate_face_bounds = face_bounds[candidate_faces_indices]
+
+    # Hardcoded the scenario when only one face is found
+    if len(candidate_faces_indices) == 1:
+        return candidate_face_data[0]
+
+    weight_df = _get_zonal_faces_weight_at_constLat(
+        candidate_face_edges_cart,
+        np.sin(np.deg2rad(constLat_deg)),
+        candidate_face_bounds,
+        is_directed=False,
+        is_latlonface=is_latlonface,
+    )
+
+    # Compute the zonal mean(weighted average) of the candidate faces
+    weights = weight_df["weight"].values
+    zonal_mean = np.sum(candidate_face_data * weights, axis=-1) / np.sum(weights)
+
+    return zonal_mean
```

Comment:
@hongyuchen1030 

Wouldn't taking the weighted average here mean that we have a Conservative Zonal Average?

Reply:
Non-conservative and conservative both take weighted averages.

Non-conservative tasks the edge length as the weight factor while the conservative ones take the area.

Reply:
So if we take the non-conservative zonal average of a face-centered data variable, the length of the edge that the line intersects would be used as the weight? 

Would this weight be applied to both faces that saddle that edge?

Reply:
“the length of the edge that the line intersects would be used as the weight?” Yes

"Would this weight be applied to both faces that saddle that edge?" Yes